### ISSUE TITLE
Updating PreTrained Model with New Examples: Preprocessing

### ISSUE THREAD

**Issue by zakraicik** [explosion/spaCy#3764]
What type of preprocessing is recommended when adding new examples to a pretrained model for named entity recognition?

**Comment by ines** [comment]
Ideally, you'd want to normalise the whitespace. Textacy is a great library for pre-processing text for spaCy: https://github.com/chartbeat-labs/textacy 

In the simplest case, the `nlp.update` methods needs two things: raw text and a list of character offsets into the text (start offset, end offset, label) describing the entity spans. You can see an example here: https://spacy.io/usage/training#example-train-ner

You probably also want to make sure that the text is close enough to the data the pre-trained models were trained on – for example, if you're using spaCy's English models, that's general news and web text.

**Comment by zakraicik** [comment]
Thanks. Will I run into problems if I train the mode directly in Jupiter as opposed to the command line?

**Comment by zakraicik** [comment]
How important is sentence structure when adding new entities to the existing NLP model? I have a lot of labeled examples, but they are in paragraph form and were provided to me stripped of punctuation.
